internal/service/instance.go

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) F5, Inc.
+ *
+ * This source code is licensed under the Apache License, Version 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package service
+
+import (
+	"github.com/nginx/agent/v3/api/grpc/instances"
+	"github.com/nginx/agent/v3/internal/datasource/nginx"
+	"github.com/nginx/agent/v3/internal/model/os"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6@v6.7.0 -generate
+//counterfeiter:generate -o mock_instance.go . InstanceServiceInterface
+//go:generate sh -c "grep -v github.com/nginx/agent/v3/internal/service mock_instance.go | sed -e s\\/service\\\\.\\/\\/g > mock_instance_fixed.go"
+//go:generate mv mock_instance_fixed.go mock_instance.go
+type InstanceServiceInterface interface {
```

Comment:
I'll just raise this point to make sure that we are making a conscious decision on this: **Do we want to propagate the convention of suffixing interfaces with** `-Interface`?

There seem to be several ways of doing this: https://medium.com/@dotronglong/interface-naming-convention-in-golang-f53d9f471593

In short:
1. Prepend the letter "I"  => `IInstanceService` - turns out this is a perfect example of a corner case where it doesn't really work too well
2. Use the `-er` suffix => `InstanceServicer` - typically used for one-method interfaces (per [Effective Go](https://go.dev/doc/effective_go#interface-names))
3. Keep it as is with the `Interface` suffix.

I don't really like any of the three tbh but I'd maybe propose suffixing `-Interface` when there's multiple methods and using `-er` if there's one method?

Reply:
whatever we decide, we should start documenting decisions like these